src/libraries/System.Net.Sockets/src/System/Net/Sockets/Socket.cs

```diff
@@ -2428,22 +2404,18 @@ public IAsyncResult BeginReceive(IList<ArraySegment<byte>> buffers, SocketFlags
             return TaskToApm.Begin(t, callback, state);
         }
 
-        public int EndReceive(IAsyncResult asyncResult)
-        {
-            ThrowIfDisposed();
-            return TaskToApm.End<int>(asyncResult);
-        }
+        public int EndReceive(IAsyncResult asyncResult) => TaskToApm.End<int>(asyncResult);
 
         public int EndReceive(IAsyncResult asyncResult, out SocketError errorCode) =>
             EndSendReceive(asyncResult, out errorCode);
 
-        private int EndSendReceive(IAsyncResult asyncResult, out SocketError errorCode)
+        private static int EndSendReceive(IAsyncResult asyncResult, out SocketError errorCode)
         {
-            ThrowIfDisposed();
-
             if (TaskToApm.GetTask(asyncResult) is not Task<int> ti)
             {
-                throw new ArgumentException(null, nameof(asyncResult));
+                throw asyncResult is null ?
+                    new ArgumentNullException(nameof(asyncResult)) :
+                    new ArgumentException(null, nameof(asyncResult));
```

Comment:
Is this a separate fix to throw a more specific exception?

Reply:
Yes, this is to match the behavior of methods which simply delegate to `TaskToApm`:

https://github.com/dotnet/runtime/blob/73cf30dd4a00a1ee7b12a82a2cff99b1870a3c4e/src/libraries/Common/src/System/Threading/Tasks/TaskToApm.cs#L66-L71